### PR TITLE
Update code style guide on toUnit function naming

### LINKED ISSUE #400: Unify Function Naming for Converting From/To Classes From Various Units
### Description of the task
We have several classes like `Angle` and `TimeStamp` that have functions like `ofDegrees` and `fromSeconds`, and in some cases have output functions like `toDegrees` or `getSeconds`, we should unify these to have a consistent naming scheme.

### Acceptance criteria

- [ ] All "input" functions are renamed to be consistent
- [ ] All "output" functions are renamed to be consistent

### PR DESCRIPTION
**Pull request by jonathanlew**
### Description
Update code style guide on toUnit function naming

### Resolved Issues
resolves  #400

### Review Checklist

**_It is the reviewers responsibility to also make sure every item here has been covered_**

- [ ] **Function & Class comments**: All function definitions (usually in the `.h` file) should have a javadoc style comment at the start of them. For examples, see the functions defined in `thunderbots/software/geom`. Similarly, all classes should have an associated Javadoc comment explaining the purpose of the class.
- [ ] **Remove all commented out code**
- [ ] **Remove extra print statements**: for example, those just used for testing
- [ ] **Resolve all TODO's**: All `TODO` (or similar) statements should either be completed or associated with a github issue
- [ ] **Justify drops in code coverage**: If this PR results in a non-trivial drop in code coverage (a bot should post a coverage diagram as a comment), please justify why we can't test the code that's not covered.

### CODE REVIEW

File: docs/code-style-guide.md [hunk 1]
```diff
@@ -225,6 +225,7 @@ Some general guidelines when writing tests are:
 * in general, getter and setter methods on classes should be written like `getName()`, `setName(string name)`, with the following exceptions
   * getters with the return type `bool` may be prefixed with `is` instead of `get`, ie. `bool isActive()`
   * getters that are used _incredibly_ frequently and are _incredibly_ obvious may not require the `get` prefix. For example `Point::x()` and `Point::y()` 
+  * getters that implicitly convert units should be written as `toUnit`. For example `Angle::toDegrees()`
```
Comment (akhilveeraghanta) [inline docs/code-style-guide.md]: uber nit: wouldn't it be explicit?

**Review by jsn0716 — DISMISSED** [review]

File: docs/code-style-guide.md (hunk 1 above)
Comment (jonathanlew) [reply docs/code-style-guide.md]: done

**Review by akhilveeraghanta — DISMISSED** [review]
LGTM :ascension_island:

**Review by jsn0716 — DISMISSED** [review]

File: docs/code-style-guide.md [hunk 2]
```diff
@@ -225,6 +225,7 @@ Some general guidelines when writing tests are:
 * in general, getter and setter methods on classes should be written like `getName()`, `setName(string name)`, with the following exceptions
   * getters with the return type `bool` may be prefixed with `is` instead of `get`, ie. `bool isActive()`
   * getters that are used _incredibly_ frequently and are _incredibly_ obvious may not require the `get` prefix. For example `Point::x()` and `Point::y()` 
+  * getters that convert units should be written as `toUnit`. For example `Angle::toDegrees()`
```
Comment (MathewMacDougall) [inline docs/code-style-guide.md]: Just double checking, is our usage of `fromType` consistent for creation as well? This should also be part of the style guide.

**Review by MathewMacDougall — CHANGES_REQUESTED** [review]

File: docs/code-style-guide.md (hunk 2 above)
Comment (jonathanlew) [reply docs/code-style-guide.md]: Yes, but we don't have a section on constructors. Should I make one?

Comment (MathewMacDougall) [reply docs/code-style-guide.md]: I think that would be good. It seems like the ticket covers creation methods / constructors

**Review by garethellis0 — DISMISSED** [review]

File: docs/code-style-guide.md (hunk 2 above)
Comment (jonathanlew) [reply docs/code-style-guide.md]: done

**Review by akhilveeraghanta — APPROVED** [review]

**Review by MathewMacDougall — APPROVED** [review]

**Review by jsn0716 — APPROVED** [review]

**Pull request merged**